### PR TITLE
Adding autocomplete to params using x-data-threescale-name field

### PR DESCRIPTION
**Pull request by damianpm**
**What this PR does / why we need it**:

Implements autocomplete of API keys, using the `x-data-threescale-name` field (see [docs](https://access.redhat.com/documentation/en-us/red_hat_3scale_api_management/2.8/html/providing_apis_in_the_developer_portal/create-activedocs-spec#activedocs_auto_fill_of_api_keys))
This PR is a continuation of the job started in https://github.com/3scale/porta/pull/1935, later reverted in https://github.com/3scale/porta/pull/1989

**Which issue(s) this PR fixes** 

https://issues.redhat.com/browse/THREESCALE-4857

**Verification steps** 
- Go to a Service's active docs section: apiconfig/services/2/api_docs
- Create a new OAS v3 using the `x-data-threescale-name` fields, as described [here](https://access.redhat.com/documentation/en-us/red_hat_3scale_api_management/2.8/html/providing_apis_in_the_developer_portal/create-activedocs-spec#activedocs_auto_fill_of_api_keys). You can use this [example](https://gist.github.com/damianpm/9c98e6710994162474a5171cd326465b)
- Add a new page or replace the default **Documentation** page with the following snippet (see [docs](https://access.redhat.com/documentation/en-us/red_hat_3scale_api_management/2.8/html/providing_apis_in_the_developer_portal/using-oas30#configuring_the_developer_portal_with_oas_3_0)):
```
{% content_for javascripts %}
    {{ 'active_docs.js' | javascript_include_tag }}
{% endcontent_for %}

<h1>Documentation</h1>

<div class="swagger-section">
  <div id="message-bar" class="swagger-ui-wrap">&nbsp;</div>
  <div id="swagger-ui-container" class="swagger-ui-wrap"></div>
</div>

<script type="text/javascript">
  (function () {
    var url = "{{provider.api_specs.first.url}}";

    SwaggerUI({ url: url, dom_id: "#swagger-ui-container" });
  }());
</script>
```

**Special notes for your reviewer**:

To switch from `user_key` to `app_id` / `app_key`, authentication, go to the edit integration settings page (/apiconfig/services/2/integration) and select an authentication method:
edit-settings
____
**Admin Portal:**

admin-user-key
____
admin-app-id
____

**Developer Portal:**

developer-user-key
____
developer-app-id

### CODE REVIEW

File: app/javascript/packs/active_docs.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 // We can define the 3scale plugins here and export the modified bundle
 import SwaggerUI from 'swagger-ui'
+import { autocompleteOAS3 } from './OAS3Autocomplete'
 import 'swagger-ui/dist/swagger-ui.css'
-
 import 'ActiveDocs/swagger-ui-3-patch.scss'
 
+window.autocompleteOAS3 = autocompleteOAS3
 window.SwaggerUI = SwaggerUI
```
Comment (josemigallas) [inline app/javascript/packs/active_docs.js:8]:
```suggestion
const accountDataUrl = '/api_docs/account_data.json'

window.SwaggerUI = (args) => {
  SwaggerUI({
    ...args,
    responseInterceptor: response => autocompleteOAS3(response, accountDataUrl)
  }
}
```

File: app/javascript/packs/active_docs.js [hunk 2]
```diff
@@ -1,7 +1,15 @@
 // We can define the 3scale plugins here and export the modified bundle
 import SwaggerUI from 'swagger-ui'
+import { autocompleteOAS3 } from './OAS3Autocomplete'
 import 'swagger-ui/dist/swagger-ui.css'
-
 import 'ActiveDocs/swagger-ui-3-patch.scss'
 
-window.SwaggerUI = SwaggerUI
+window.autocompleteOAS3 = autocompleteOAS3
```
Comment (josemigallas) [inline app/javascript/packs/active_docs.js:7]:
```suggestion
```

File: app/javascript/src/Types/SwaggerTypes.jsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
+export type Primitive = string | number | boolean
+
+export type AccountData = {
+ [string]: Array<{name: string, value: string}>
```
Comment (josemigallas) [inline app/javascript/src/Types/SwaggerTypes.jsx:4]:
```suggestion
 [string]: Array<{ name: string, value: string }>
```
Like in https://github.com/3scale/porta/pull/1991/files#diff-186a85894575b636b4b537b9c7472c57R13

File: spec/javascripts/ActiveDocs/ActiveDocsSpec.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import SwaggerUI from 'swagger-ui-react'
+import { ActiveDocsSpec } from 'ActiveDocs/components/ActiveDocsSpec'
+
+const url = 'foo.example.com'
+const responseInterceptor = jest.fn(() => () => {})
+
+it('should render itself', () => {
+  const wrapper = mount(<ActiveDocsSpec url={url} responseInterceptor={responseInterceptor} />)
+  expect(wrapper.find(ActiveDocsSpec).exists()).toBe(true)
+  expect(wrapper).toMatchSnapshot()
```
Comment (josemigallas) [inline spec/javascripts/ActiveDocs/ActiveDocsSpec.spec.jsx:14]: I think this snapshot is not really meaningful.
```suggestion
```

File: spec/javascripts/ActiveDocs/ActiveDocsSpec.spec.jsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+})
+
+it('should render swagger-ui', () => {
+  const wrapper = mount(<ActiveDocsSpec url={url} responseInterceptor={responseInterceptor}/>)
+  expect(wrapper.find('.swagger-ui').exists()).toBe(true)
+  expect(wrapper.find(SwaggerUI).prop('url')).toBe('foo.example.com')
+  expect(typeof wrapper.find(SwaggerUI).prop('responseInterceptor')).toBe('function')
```
Comment (josemigallas) [inline spec/javascripts/ActiveDocs/ActiveDocsSpec.spec.jsx:21]:
```suggestion
  expect(responseInterceptor).toHaveBeenCalled()
```

File: app/javascript/packs/active_docs.js (hunk 2 above)
Comment (damianpm) [reply app/javascript/packs/active_docs.js:7]: 🤦

File: app/javascript/packs/service_active_docs.js [hunk 6]
```diff
@@ -1,8 +1,18 @@
...
+  const DATA_URL = 'p/admin/api_docs/account_data.json'
+  const { url, baseUrl } = document.getElementById(containerId).dataset
+  const accountDataUrl = `${baseUrl}${DATA_URL}`
 
-  ActiveDocsSpec({ url }, containerId)
+  const responseInterceptor = (response) => autocompleteOAS3(response, accountDataUrl)
+
+  SwaggerUI({
+    url,
+    dom_id: '#swagger-ui-container',
```
Comment (josemigallas) [inline app/javascript/packs/service_active_docs.js:15]:
```suggestion
    dom_id: `#${containerId}`,
```

**Pull request merged**